### DISCUSSION TITLE
Bulk insert `deliver_by :database`

### DISCUSSION THREAD

**Discussion by khrisnagunanasurya-iav** [excid3/noticed#335] [General]
Hi, is it possible to bulk insert to the database? as of right now when I tried to send a notification to 5 users, there are 5 insert queries, and if there are 10000 users, I think it will execute 10000 queries, so if possible I want to bulk the insert to the database, let's say 1000 users each query.

I already tried using `deliver_later`, but it only does execute the send notification in the background.

I already read [about this](https://github.com/excid3/noticed/issues/89), but it's mentioning that even though it's a batch delivery for `Postmark`, it still executes deliver to database 5.000 times

**Comment by eserna27** [comment]
Hi @khrisnagunanasurya-iav, I am trying to solve this problem in this pull request eserna27#1, if you have some comments, I would be glad to receive/answer them.

**Comment by excid3** [comment]
Yeah, we should definitely bulk insert the to the database. That's separate than bulk delivery for other methods like email (some delivery methods may support it, but so may not).
